code/mob/living/critter/flockcritter_parent.dm

```diff
@@ -541,5 +541,5 @@
 		amounttopay = min(F.resources, difference, 10)
 		F.pay_resources(amounttopay)
 		target.currentmats += amounttopay
-		if(F.resources && !F.is_npc) //npc check just to make sure it doesnt interfere with their ai.
+		if(F.resources)
```

Comment:
Will a flockdrone keep depositing resources if it's attacked, say?

Reply:
the action bar should be interruptible by movement/actions/stun and so any AI task that does any of those will interrupt the deposit. I haven't actually tested that scenario though - it should work.